test/apps/iodemo/io_demo.cc

```diff
@@ -948,6 +948,21 @@ class P2pDemoCommon : public UcxContext {
         validate(conn, msg, iomsg_size);
     }
 
+    static inline void check_conn_id(const iomsg_t *msg, UcxConnection *conn)
+    {
+#ifndef NDEBUG
```

Comment:
no need NDEBUG because this is part of data integrity test which is already under runtime `validate` condition, also I think we should abort instead of `assert` below

Reply:
removed